### PR TITLE
feat(carousel): add thumbs option

### PR DESCRIPTION
**Pull request by cwklausing**
[Card](https://rentpath.atlassian.net/browse/AL-35)

Creates prop that gives ability to add thumbnail previews to `react-image-gallery`.

This PR is a dependency for [this](https://github.com/rentpath/ag.js/pull/5648) PR

### CODE REVIEW

**Comment by mpelzsherman** [comment]
Looks good! Do we need to add any tests?

**Comment by cwklausing** [comment]
I'm not sure... since we're just passing props (which default to false), I'm inclined to say no. But would also be easy to add a quick prop check if we were interested.

**Comment by tadjohnston** [comment]
Why is this needed? From the looks of it, `...rest` should handle this without needing to add the right right?

**Comment by tadjohnston** [comment]
Why is this needed? From the looks of it, `...rest` should handle this without needing to add the right right? You would just need to pass `showThumbnails` instead.

File: packages/react-ui-core/src/Carousel/Carousel.js [hunk 1]
```diff
@@ -212,11 +215,12 @@ export default class Carousel extends Component {
           items={this.items}
           renderItem={this.renderItem}
           onSlide={this.onSlide}
-          showThumbnails={false}
+          showThumbnails={showThumbs}
```
Comment (barelyfitz) [inline packages/react-ui-core/src/Carousel/Carousel.js]: Why not name the prop `showThumbs`? Seems like having two different names just confuses things.

Comment (cwklausing) [reply packages/react-ui-core/src/Carousel/Carousel.js]: fair enough--I think that makes sense.

File: packages/react-ui-core/src/Carousel/Carousel.js [hunk 2]
```diff
@@ -212,11 +215,12 @@ export default class Carousel extends Component {
...
           showPlayButton={false}
           showFullscreenButton={false}
           startIndex={selectedIndex}
           infinite={false}
+          lastThumbnailOffset={showThumbs ? 3 : null}
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Carousel/Carousel.js]: What if we wanted 4? How does that work here?

**Comment by mpelzsherman** [comment]
@tadjohnston I get the reluctance to add unnecessary props, but I think a `showThumbnails` prop makes sense on `Carousel`. Exposing props in the component where they are most likely to be used is a lot more developer-friendly.

**Comment by cwklausing** [comment]
In light of Tad's comments about the `...rest`, I think we can achieve anything we need here without altering React-UI

**Comment by tadjohnston** [comment]
@mpelzsherman I disagree. It's not ideal to keep adding one-off props to appease a use case in one repo.  The fact that the component here can already handle the prop in question means it's flexible enough to handle what's thrown at it which was the original intention of writing components in this library in this manner.  If we were to add a prop in every situation, this would become a mess and also require a publish / release every single time.  

These are all the props react-image-gallery currently takes in:
```
    flickThreshold: PropTypes.number,
    items: PropTypes.array.isRequired,
    showNav: PropTypes.bool,
    autoPlay: PropTypes.bool,
    lazyLoad: PropTypes.bool,
    infinite: PropTypes.bool,
    showIndex: PropTypes.bool,
    showBullets: PropTypes.bool,
    showThumbnails: PropTypes.bool,
    showPlayButton: PropTypes.bool,
    showFullscreenButton: PropTypes.bool,
    disableThumbnailScroll: PropTypes.bool,
    disableArrowKeys: PropTypes.bool,
    disableSwipe: PropTypes.bool,
    useBrowserFullscreen: PropTypes.bool,
    preventDefaultTouchmoveEvent: PropTypes.bool,
    defaultImage: PropTypes.string,
    indexSeparator: PropTypes.string,
    thumbnailPosition: PropTypes.string,
    startIndex: PropTypes.number,
    slideDuration: PropTypes.number,
    slideInterval: PropTypes.number,
    swipeThreshold: PropTypes.number,
    swipingTransitionDuration: PropTypes.number,
    onSlide: PropTypes.func,
    onScreenChange: PropTypes.func,
    onPause: PropTypes.func,
    onPlay: PropTypes.func,
    onClick: PropTypes.func,
    onImageLoad: PropTypes.func,
    onImageError: PropTypes.func,
    onTouchMove: PropTypes.func,
    onTouchEnd: PropTypes.func,
    onTouchStart: PropTypes.func,
    onMouseOver: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onThumbnailError: PropTypes.func,
    onThumbnailClick: PropTypes.func,
    renderCustomControls: PropTypes.func,
    renderLeftNav: PropTypes.func,
    renderRightNav: PropTypes.func,
    renderPlayPauseButton: PropTypes.func,
    renderFullscreenButton: PropTypes.func,
    renderItem: PropTypes.func,
    stopPropagation: PropTypes.bool,
    additionalClass: PropTypes.string,
    useTranslate3D: PropTypes.bool,
    isRTL: PropTypes.bool
```

So where do you stop in adding these? Is it very developer friendly to spread props like we're doing? Probably not, but I don't see how that's any different than having to track a prop (prop drilling) through multiple levels possibly.

Unless you're specifically adding logic for the prop coming in or you're doing something weird with props where you need to desctructure it, there generally isn't a good use case to add it IMO.

**Pull request closed without merging**